### DISCUSSION TITLE
Add calendar

### DISCUSSION THREAD

**Discussion by astashov** [astashov/liftosaur#127] [Ideas]
Currently you can only see your history as a flat list.
Add calendar view, that would show your workouts on a calendar.

**Comment by Posdiver** [comment]
What I really liked, coming from another app, was to see how many workouts/volume I did over the past 7 days. This helps making sure you accomplish your goals every week. This is at a glance when you open up the app. It could be under Workout history, on top of the page. The section where you can start the next workout could be collapsed (workout details), thus saving space for this graph...
 
![image](https://github.com/user-attachments/assets/355d2ab4-4f1e-4d71-88c4-85984ce691c8)

Also, it had a calendar that I mostly used just to see how constant I was over months. This is also what I would need. The app uses a color scheme to help distinguish workouts, it may or may not be necessary here...  
![image](https://github.com/user-attachments/assets/431421c4-b605-4900-afae-8292d619a4d7)

**Comment by winder** [comment]
I like to use a calendar primarily as a way to see how consistent I've been. For that use case simple dots overlaid on the calendar is fine. Being able to smoothly scroll through the months is also nice. Selecting a dot to jump to a specific workout in the list view is also nice, since it's what I'd expect it to do, but I'm not sure why anyone would ever do that. 

<img src="https://github.com/user-attachments/assets/28114202-a97c-47df-8aa4-05534b402150" width="400" />

**Comment by astashov** [comment]
There was also an idea (from a user through email) to add notes to the calendar days

**Comment by TimKamppinen** [comment]
Just imported 9 years of data (from Strong via Hevy) and it's pretty much useless until there's a quicker way to access the time periods that I want to go back and review. Hope the calendar is added sooner rather than later.

**Comment by astashov** [comment]
Adding more calendar requests here:

> I have one feature request: could a simple calendar be implemented so I can see at a glance the frequency of my workouts? Something like an overview of a month, with the days on which I trained highlighted. Pressing the highlighted day could bring me to the workout performed on that day. Maybe it would not fit with the app layout or it is not straightforward to implement - it's certainly not a dealbreaker for me, but I thought a suggestion doesn't hurt :)

**Comment by astashov** [comment]
> I I have a little calendar printed out in my home gym where I just circle the days that I train so that I can get an overview of my training days really quickly. I think it would be great if you could do something similar in the app, where you have a little calendar overview of the month, and the days that you train have a little circle around them. Maybe, you could even make it so that you can tap the day that you worked out on and it opens the out of the day. I also tend to note reasons why I didn’t train, I.e. sick or vacation. That might also be a cool feature, so that maybe you could have purple for workout days, as per when you worked out, but then also add the ability of the user to tap a certain day and you could add a label. So that sick days would be grey, cardio days could be yellow etc. And then maybe you could even make it so that you can tap the day with a certain label and add a simple note, e.g. “hiking, 20 km “. I guess many people have a tendency to overtrain and that could help them visualize it.
Do you think that would be possible?

**Comment by astashov** [comment]
> One feature I can't find is a calendar view where a lifter can see what calendar days there are workouts logged (see screenshots from Strong and Fitnotes). Is this feature available on Liftosaur or is it something that can be expected in the future? Thanks in advance for your help.

| One | Two |
| --- | --- |
|![image](https://github.com/user-attachments/assets/78f5dba5-e74f-4716-92ce-cbffe22f53e7)|![image](https://github.com/user-attachments/assets/0fe10678-9566-4146-be36-419df1cb2ce6)|

**Comment by astashov** [comment]
> Can you add a calendar type feature in the graphs section so I can see the days of the month that I did a workout? I think it’d be more intuitive in conjunction with the sets/ volume per week graph.

**Comment by astashov** [comment]
> Teacoin: it would be really cool if there was a calendar feature implemented, to see which days had which workouts and what exercises
> KillerK009: Yea would be really cool to see a history view that's similar to the program preview but it's just calendar days/weeks and shows the workouts done on those days so you can review previous weeks and sessions all at a glance!
> Teacoin: Yeah, this is what I had in mind, displayed in calendar format. Fitnotes has a similar feature and it's pretty good.

**Comment by TimKamppinen** [comment]
Just one additional piece of input, I think the calendar implementation like in Strong where you can see more than one month at a time and scroll back through the months/years is VASTLY superior to some other apps I have been testing where you can only see one month at a time. Even worse, some of them don't even have a "month picker" so if you wanted to see what you were doing a year ago you would have to tap the little arrow 12 times. The scrolling calendar is faster to get to a particular point in the past and also allows you to scan through and get a picture of your overall trends in consistency across longer time frames.

**Comment by AdEmergency625** [comment]
I would like to the calendar to jump back to a particular day to see the workout I did that day. Ideally we will also have a duplicate button too. Use case, go back to the same gym (like the same hotel), it would be super cool to not having had to enter everything again. 
another one is to pick a particular exercise at that day and add to current workout. Because hotel gyms tend to have limited equipment. So I ended up pick and choose from past workouts to create the workout (better than creating from scratch using the ad học workout)

**Comment by StanleyMitchell8** [comment]
How difficult would it be to just allow the app to sync with our calendar such as google calendar etc, similar to the way that it syncs to apple health so that when we complete a workout it automatically syncs that to our calendar of choice and titles it whatever our workout was for that day and time, and includes a link to the workout in the description on the calendar so we have easy access back to it in the app?

**Comment by astashov** [comment]
This is done!

**Comment by StanleyMitchell8** [comment]
Awesome news!!! Thank you!!On Feb 10, 2025, at 11:59 AM, Anton Astashov ***@***.***> wrote:﻿
Closed #127 as resolved.

**Comment by Posdiver** [comment]
This is awesome! Thank you so much
